external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTest.scala

```diff
@@ -29,4 +29,13 @@ trait KafkaTest extends BeforeAndAfterAll {
     super.afterAll()
     CachedKafkaProducer.clear()
   }
+
+  def waitForBiggerTimestamp(curTimestamp: Long): Long = {
```

Comment:
AFAIK timestamp field can be overridden in the `ProducerRecord`, can't be? Looks like this part generates a lot of unit test wait.

Reply:
Ah yes nice suggestion. Will address.